docs/aws/audit/elbmonitoring/rules/elb_cross_zone_enabled.mdx

```diff
@@ -24,6 +24,224 @@ NIST, SOC2, GDPR, HITRUST, NISTCSF
+   - Click on the "Edit attributes" button.
+   - In the "Edit load balancer attributes" dialog, check the box for "Cross-Zone Load Balancing."
+   - Click "Save" to apply the changes.
+
+By following these steps, you ensure that Cross-Zone Load Balancing is enabled for your ELBs, which helps in distributing traffic evenly across all registered instances in all enabled Availability Zones.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Elastic Load Balancers (ELBs) from having Cross-Zone Load Balancing disabled using AWS CLI, follow these steps:
+
+1. **Identify the Load Balancer:**
+   First, list all your load balancers to identify the one you want to configure.
+
+   ```sh
+   aws elb describe-load-balancers
+   ```
+
+2. **Check Current Cross-Zone Load Balancing Setting:**
+   Verify the current cross-zone load balancing setting for your load balancer.
+
+   ```sh
+   aws elb describe-load-balancer-attributes --load-balancer-name <your-load-balancer-name>
+   ```
+
+3. **Enable Cross-Zone Load Balancing:**
+   Enable cross-zone load balancing for the identified load balancer.
+
+   ```sh
+   aws elb modify-load-balancer-attributes --load-balancer-name <your-load-balancer-name> --load-balancer-attributes "{\"CrossZoneLoadBalancing\":{\"Enabled\":true}}"
+   ```
+
+4. **Verify the Change:**
+   Confirm that cross-zone load balancing has been enabled.
+
+   ```sh
+   aws elb describe-load-balancer-attributes --load-balancer-name <your-load-balancer-name>
+   ```
+
+By following these steps, you can ensure that cross-zone load balancing is enabled for your Elastic Load Balancers using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Elastic Load Balancers (ELBs) from being created without cross-zone load balancing enabled in AWS using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+       aws_secret_access_key='YOUR_AWS_SECRET_KEY',
+       region_name='YOUR_AWS_REGION'
+   )
+   ```
+
+3. **Create ELB Client**:
+   Create an ELB client using the session.
+
+   ```python
+   elb_client = session.client('elb')
+   ```
+
+4. **Create ELB with Cross-Zone Load Balancing Enabled**:
+   When creating an ELB, ensure that the `CrossZoneLoadBalancing` attribute is set to `True`.
+
+   ```python
+   response = elb_client.create_load_balancer(
+       LoadBalancerName='my-load-balancer',
+       Listeners=[
+           {
+               'Protocol': 'HTTP',
+               'LoadBalancerPort': 80,
+               'InstanceProtocol': 'HTTP',
+               'InstancePort': 80,
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>